### ISSUE TITLE
New rule: Warn when sort is called with no arguments on array of non-string types

### ISSUE THREAD

**Issue by fiesh** [typescript-eslint/typescript-eslint#1806]
This is https://github.com/eslint/eslint/issues/13098 where it was deemed not suitable due to lacking type information.

To summarize what was said there, the following behavior seems desirable to me:
```ts
[5, 0, 10].sort(); // Warn
["5", "0", "10"].sort(); // Don't warn
[5, 0, 10].sort(somePredicate); // Don't warn
[5, 0, 10].sort((lhs, rhs) => lhs - rhs); // Don't warn
```
The reason is that calling `sort` without a predicate implicitly converts all elements of an array to `string` before sorting, leading to potentially surprising results.

**Comment by fiesh** [comment]
Oh wow, it already exists!  Sorry, don't know how I missed it.  I think the current behavior is fine, it reporting on string arrays is really just a minor issue.  Thank you!
